GithubBrowserSample/app/src/main/java/com/android/example/github/binding/FragmentBindingAdapters.kt

```diff
@@ -34,4 +37,22 @@ class FragmentBindingAdapters @Inject constructor(val fragment: Fragment) {
     fun bindImage(imageView: ImageView, url: String?) {
         Glide.with(fragment).load(url).into(imageView)
     }
+
+    @BindingAdapter("sharedElementImageUrl")
+    fun bindSharedImage(imageView: ImageView, url: String?) {
```

Comment:
it is really scary to do this kind of logic in a binding a binding adapter since it has a side effect on the fragment itself (which is not even included in the binding). e.g. what happens if i have 2 of these?

I think this falls into the part of abusive binding adapter usage :) . I would much prefer to do this explicitly in the fragment. I'm also worried that people will take it as an example of Binding adapters usage.

After discussing offline w/ Ian, here is a recommendation:

* change the `bindImage` adapter above to receive an optional `completedCallback`. Then it can just call the completed callback if it exists.
* change the code in the fragment to provide a `callback` to the `Binding` which is bound to the related image view; then do the startPostpone in the fragment when callback is received.

This makes the Binding adapter generic, w/o side effects. There is more code in the fragment but it should be because fragment is the one which stopped the transition at the first place.

Also, this code will stall if network is bad. The implementation in the fragment should have a timeout (1 s max?)